airflow/dag_processing/bundles/git.py

```diff
@@ -84,6 +94,13 @@ def _process_git_auth_url(self):
         elif not self.repo_url.startswith("git@") or not self.repo_url.startswith("https://"):
             self.repo_url = os.path.expanduser(self.repo_url)
 
+    def _setup_inline_key(self):
+        with tempfile.NamedTemporaryFile(mode="w", delete=False) as tmp_keyfile:
```

Comment:
We might want to refactor things a bit.

- We shouldn't leave this fine around imo, so let's do a context manager so we can clean up.
- This likely means the way we do env likely needs to be adjusted a bit too.
- This also means we won't have to write the pk out when creating a bundle just to do the view_url (e.g. on the webserver).

Reply:
I've completed the refactoring based on your comments.